src/browser/fetch/request.rs

```diff
@@ -0,0 +1,282 @@
+//! The Request of the Fetch API.
+
+use super::{header, FetchError, Header, Headers, Method, Result};
+use gloo_timers::callback::Timeout;
+use serde::Serialize;
+use std::{borrow::Cow, cell::RefCell, rc::Rc};
+use wasm_bindgen::JsValue;
+
+/// Its methods configure the request, and handle the response. Many of them return the original
+/// struct, and are intended to be used chained together.
+#[derive(Debug, Clone, Default)]
+pub struct Request<'a> {
+    url: Cow<'a, str>,
+    headers: Headers<'a>,
+    method: Method,
+    body: Option<JsValue>,
+    cache: Option<web_sys::RequestCache>,
+    credentials: Option<web_sys::RequestCredentials>,
+    integrity: Option<String>,
+    mode: Option<web_sys::RequestMode>,
+    redirect: Option<web_sys::RequestRedirect>,
+    referrer: Option<String>,
+    referrer_policy: Option<web_sys::ReferrerPolicy>,
+    timeout: Option<u32>,
+    controller: RequestController,
+}
+
+impl<'a> Request<'a> {
+    /// Create new request based on the provided url.
+    ///
+    /// To get a [`Response`](./struct.Response.html) you need to pass
+    /// `Request` to the [`fetch`](./fn.fetch.html) function.
+    ///
+    /// [MDN reference](https://developer.mozilla.org/en-US/docs/Web/API/Request)
+    pub fn new(url: impl Into<Cow<'a, str>>) -> Self {
+        Self {
+            url: url.into(),
+            ..Self::default()
+        }
+    }
+
+    // TODO: remove when https://github.com/rust-lang/rust-clippy/issues/4979 will be fixed
+    #[allow(clippy::missing_const_for_fn)]
+    /// Set headers for this request.
+    /// It will replace any existing headers.
+    pub fn headers(mut self, headers: Headers<'a>) -> Self {
+        self.headers = headers;
+        self
+    }
+
+    /// Set specific header.
+    pub fn header(mut self, header: Header<'a>) -> Self {
+        self.headers.set(header);
+        self
+    }
+
+    /// Set HTTP method. Default method is `GET`.
+    pub const fn method(mut self, method: Method) -> Self {
+        self.method = method;
+        self
+    }
+
+    /// Set request body to provided `JsValue`. Consider using `json` or `text` methods instead.
+    ///
+    /// Note that a request using the GET or HEAD method cannot have a body.
```

Comment:
Add a note or `/// # Panics` block, please, so users know that it panics when you set body and use GET or HEAD.